win32ss/user/ntuser/sysparams.c

```diff
@@ -1404,10 +1404,10 @@ SpiGetSet(UINT uiAction, UINT uiParam, PVOID pvParam, FLONG fl)
             return SpiGetInt(pvParam, &gspv.bFontSmoothing, fl);
 
         case SPI_SETFONTSMOOTHING:
-            gspv.bFontSmoothing = (uiParam == 2);
+            gspv.bFontSmoothing = uiParam;
             if (fl & SPIF_UPDATEINIFILE)
             {
-                SpiStoreSz(KEY_DESKTOP, VAL_FONTSMOOTHING, (uiParam == 2) ? L"2" : L"0");
+                SpiStoreSzInt(KEY_DESKTOP, VAL_FONTSMOOTHING, gspv.bFontSmoothing ? 2 : 0);
```

Comment:
Doesn't the "2" value imply that there should be two algorithms of font smoothing?

Reply:
Here the "2" seems to be an internal value (stored by the win32k) that is not directly related to the `FE_FONTSMOOTHING***` defined values. (Those latter ones are for the `SPI_GET/SETFONTSMOOTHINGTYPE` messages.)

Reply:
Perhaps at some point in Window's development Microsoft considered using the font smoothing registry value to also store the font smoothing type, but after my own investigation I only ever found it to be either 2 or 0. Since SPI_SETFONTSMOOTHING is only supposed to take in a Boolean I felt this was the right way to do it. Feel free to check on your own Windows installs, the registry key is in `HKCU\Control Panel\Desktop\FontSmoothing`.